src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp

```diff
@@ -0,0 +1,183 @@
+template <typename InterpolationTargetTag, typename Frame>
+struct ApparentHorizon {
+  using options_type = OptionHolders::ApparentHorizon<Frame>;
+  using const_global_cache_tags = tmpl::list<InterpolationTargetTag>;
+  using initialization_tags =
+      tmpl::append<StrahlkorperTags::items_tags<Frame>,
+                   tmpl::list<::Tags::FastFlow, ::Tags::Verbosity>,
+                   StrahlkorperTags::compute_items_tags<Frame>>;
+  template <typename DbTags, typename Metavariables>
+  static auto initialize(
+      db::DataBox<DbTags>&& box,
+      const Parallel::ConstGlobalCache<Metavariables>& cache) noexcept {
+    const auto& options = Parallel::get<InterpolationTargetTag>(cache);
+
+    // Put Strahlkorper and its ComputeItems, FastFlow,
+    // and verbosity into a new DataBox.
+    return db::create_from<
+        db::RemoveTags<>,
+        db::AddSimpleTags<tmpl::push_back<StrahlkorperTags::items_tags<Frame>,
+                                          ::Tags::FastFlow, ::Tags::Verbosity>>,
+        db::AddComputeTags<StrahlkorperTags::compute_items_tags<Frame>>>(
+        std::move(box), std::move(options.strahlkorper),
+        std::move(options.fast_flow), options.verbosity);
+    // Note: the above std::moves things out of cache. Should be ok because
+    // we don't reuse those cache items.
+  }
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent,
+            Requires<tmpl::list_contains_v<
+                DbTags, typename Tags::TemporalIds<Metavariables>>> = nullptr>
+  static void apply(
+      db::DataBox<DbTags>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::ConstGlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/,
+      const typename Metavariables::temporal_id& temporal_id) noexcept {
+    const auto& fast_flow = db::get<::Tags::FastFlow>(box);
+    const auto& strahlkorper =
+        db::get<StrahlkorperTags::Strahlkorper<Frame>>(box);
+
+    const size_t L_mesh = fast_flow.current_l_mesh(strahlkorper);
+    const auto prolonged_strahlkorper =
+        Strahlkorper<Frame>(L_mesh, L_mesh, strahlkorper);
+
+    // Make a DataBox so we can get coords from prolonged_strahlkorper
+    const auto prolonged_box = db::create<
+        db::AddSimpleTags<StrahlkorperTags::items_tags<Frame>>,
+        db::AddComputeTags<StrahlkorperTags::compute_items_tags<Frame>>>(
+        std::move(prolonged_strahlkorper));
+
+    // In the future, when we add support for multiple Frames,
+    // the code that transforms coordinates from the Strahlkorper Frame
+    // to `Metavariables::domain_frame` will go here.  That transformation
+    // may depend on `temporal_id`.
+    send_points_to_interpolator<InterpolationTargetTag>(
```

Comment:
it would be clearer that the comment does not describe line 174 with an additional newline.